infra/gp-multena/templates/02-configmaps.yaml

```diff
@@ -0,0 +1,58 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "gp-multena.name" . }}-config
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "gp-multena.labels" . | nindent 4 }}
+data:
+  config.yaml: |-
+      version: 2
+      proxy:
+        log_level: {{ .Values.multena.logLevel }}
+        provider: {{ .Values.multena.provider}}
+        thanos_url: {{ .Values.multena.thanosUrl }}
+        loki_url: {{ .Values.multena.lokiUrl }}
+        prom_label_url: {{ .Values.multena.promLabelUrl }}
+        jwks_cert_url: {{ .Values.multena.jwksCertUrl}}
+        tenant_label: {{ .Values.multena.tenantLabel}}
+        admin_group: {{ .Values.multena.adminGroup }}
+        port: 8080
+      {{- if eq .Values.multena.provider "mysql" }}
+      db:
+        enabled: true
+        user: {{ .Values.multena.db.user }}
+        password_path: /etc/secrets/db/password
+        host: {{ .Values.multena.db.host }}
+        port: {{ .Values.multena.db.port }}
+        db_name: {{ .Values.multena.db.dbName }}
+        query: {{ .Values.multena.db.query }}
+      {{- end }}
+      dev:
+        enabled: {{ .Values.multena.dev }}
+---
+{{- if eq .Values.multena.provider "configmap" }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "gp-multena.name" . }}-groups
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "gp-multena.labels" . | nindent 4 }}
+data:
+  groups.yaml: |-
```

Comment:
hier fehlt mir vlt. ein wenig das verständnis über die funktionsweise des proxies. aber die configmap wird ja nur von der application selber befüllt und ausgelesen, oder? dann würde ich kein example reingeben. 

wie geht der proxy allgemein um mit änderungen in der CM? reloaded er? liest er sich initial alle daten in den memory und geht dann nur noch darauf? bzw. schreibt dann zurück, wenn etwas dazukommt? gibt's einen konfig reload?

Reply:
Multena-proxy befüllt gar keine configmaps; -> Multena-rbac-collector. Die cm kann ebenfalls per Hand konfiguriert werden, falls das gewünscht ist. Wenn die CM geändert wird reloaded sich ebenfalls der Proxy, braucht aber da Kubernetes die configmaps in den pods nicht sofort updated.